### PR TITLE
[Executorch][llama] Allow custom sdpa op replacement pass to leverage attention mask

### PR DESCRIPTION
**Pull request by kimishpatel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #10286
* __->__ #10285
* #10284
* #10283

Previously we assumed that the custom sdpa always does causal attention.
This diff adds option to this module swap pass to make custom sdpa leverage
attention mask instead of causal.

Differential Revision: [D73222736](https://our.internmc.facebook.com/intern/diff/D73222736/)

### CODE REVIEW

File: examples/models/llama/source_transformation/sdpa.py [hunk 1]
```diff
@@ -47,34 +63,54 @@ def forward(
...
-        output = torch.ops.llama.custom_sdpa(
-            q,
-            k,
-            v,
-            input_pos[0].item(),
-            None,  # Attention mask
-            0,  # dropout probability. Ignored by the code
-            True,  # is_causal
-        )
+        if self.use_attention_mask:
```
Comment (lucylq) [inline examples/models/llama/source_transformation/sdpa.py:66]: nit, maybe this could be:

if not self.use_attention_mask:
    mask = None
    
output = torch.ops.llama.custom_sdpa(
                q,
                k,
                v,
                input_pos[0].item(),
                mask,  # Attention mask
                0,  # dropout probability. Ignored by the code
                False,  # is_causal
            )

**Review by lucylq — APPROVED** [review]

File: examples/models/llama/source_transformation/sdpa.py (hunk 1 above)
Comment (kimishpatel) [reply examples/models/llama/source_transformation/sdpa.py:65]: Note that is_causal is True in one case, while not true in the other
